PiwikTracker/PiwikUserDefaults.swift

```diff
@@ -5,8 +5,11 @@ import Foundation
 /// All getter and setter are sideeffect free and automatically syncronize
 /// after writing.
 internal struct PiwikUserDefaults {
-    static var standard = PiwikUserDefaults()
-    let userDefaults = UserDefaults.standard
+    let userDefaults: UserDefaults
+
+    init(suiteName: String?) {
```

Comment:
Your properties could then return non-optional `Date` and `ClientId` by setting a default value when user defaults return nil. This will make the caller code more easier to write without any `guards` and `if let`

Could consider reaming the struct to hide the fact that it is backed by `NSUserDefaults` implementation.